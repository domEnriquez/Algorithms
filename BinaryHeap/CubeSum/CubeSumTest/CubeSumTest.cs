using ApprovalTests;
using ApprovalTests.Reporters;
using CubeSum;
using NUnit.Framework;
using System;
using System.IO;
using System.Text;

namespace CubeSumTest
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class CubeSumTest
    {
        CubeSumBinaryHeap cs;
        StringBuilder fakeOutput;

        [SetUp]
        public void SetUp()
        {
            cs = new CubeSumBinaryHeap();
            fakeOutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeOutput));
        }

        [Test]
        public void showSortedCubeSumOf0()
        {
            cs.ShowCubeSums(0);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumOf0To2()
        {
            cs.ShowCubeSums(2);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumof0To5()
        {
            cs.ShowCubeSums(5);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showSortedCubeSumof0to12()
        {
            cs.ShowCubeSums(12);

            Approvals.Verify(fakeOutput);
        }

        //All distinct integers a, b, c, and d between 0 and 10^6 such that a^3 + b^3 = c^3 + d^3
        [Test]
        public void showCubeSumsObtainedInTwoWays() 
        {
            cs.CubeSumsObtainedInTwoWays(12);

            Approvals.Verify(fakeOutput);
        }

        [Test]
        public void showCubeSumsObtainedInTwoWaysOf0To1000()
        {
            cs.CubeSumsObtainedInTwoWays(1000);

            Approvals.Verify(fakeOutput);
        }
    }
}
