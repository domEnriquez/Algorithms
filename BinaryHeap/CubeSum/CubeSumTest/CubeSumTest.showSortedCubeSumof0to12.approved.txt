0 = 0^3 + 0^3
1 = 1^3 + 0^3
1 = 0^3 + 1^3
2 = 1^3 + 1^3
8 = 2^3 + 0^3
8 = 0^3 + 2^3
9 = 1^3 + 2^3
9 = 2^3 + 1^3
16 = 2^3 + 2^3
27 = 3^3 + 0^3
27 = 0^3 + 3^3
28 = 1^3 + 3^3
28 = 3^3 + 1^3
35 = 2^3 + 3^3
35 = 3^3 + 2^3
54 = 3^3 + 3^3
64 = 4^3 + 0^3
64 = 0^3 + 4^3
65 = 1^3 + 4^3
65 = 4^3 + 1^3
72 = 2^3 + 4^3
72 = 4^3 + 2^3
91 = 3^3 + 4^3
91 = 4^3 + 3^3
125 = 5^3 + 0^3
125 = 0^3 + 5^3
126 = 1^3 + 5^3
126 = 5^3 + 1^3
128 = 4^3 + 4^3
133 = 2^3 + 5^3
133 = 5^3 + 2^3
152 = 3^3 + 5^3
152 = 5^3 + 3^3
189 = 4^3 + 5^3
189 = 5^3 + 4^3
216 = 0^3 + 6^3
216 = 6^3 + 0^3
217 = 1^3 + 6^3
217 = 6^3 + 1^3
224 = 2^3 + 6^3
224 = 6^3 + 2^3
243 = 3^3 + 6^3
243 = 6^3 + 3^3
250 = 5^3 + 5^3
280 = 4^3 + 6^3
280 = 6^3 + 4^3
341 = 5^3 + 6^3
341 = 6^3 + 5^3
343 = 7^3 + 0^3
343 = 0^3 + 7^3
344 = 1^3 + 7^3
344 = 7^3 + 1^3
351 = 2^3 + 7^3
351 = 7^3 + 2^3
370 = 3^3 + 7^3
370 = 7^3 + 3^3
407 = 4^3 + 7^3
407 = 7^3 + 4^3
432 = 6^3 + 6^3
468 = 5^3 + 7^3
468 = 7^3 + 5^3
512 = 8^3 + 0^3
512 = 0^3 + 8^3
513 = 1^3 + 8^3
513 = 8^3 + 1^3
520 = 2^3 + 8^3
520 = 8^3 + 2^3
539 = 3^3 + 8^3
539 = 8^3 + 3^3
559 = 6^3 + 7^3
559 = 7^3 + 6^3
576 = 4^3 + 8^3
576 = 8^3 + 4^3
637 = 5^3 + 8^3
637 = 8^3 + 5^3
686 = 7^3 + 7^3
728 = 6^3 + 8^3
728 = 8^3 + 6^3
729 = 0^3 + 9^3
729 = 9^3 + 0^3
730 = 1^3 + 9^3
730 = 9^3 + 1^3
737 = 2^3 + 9^3
737 = 9^3 + 2^3
756 = 3^3 + 9^3
756 = 9^3 + 3^3
793 = 4^3 + 9^3
793 = 9^3 + 4^3
854 = 5^3 + 9^3
854 = 9^3 + 5^3
855 = 8^3 + 7^3
855 = 7^3 + 8^3
945 = 6^3 + 9^3
945 = 9^3 + 6^3
1000 = 0^3 + 10^3
1000 = 10^3 + 0^3
1001 = 1^3 + 10^3
1001 = 10^3 + 1^3
1008 = 2^3 + 10^3
1008 = 10^3 + 2^3
1024 = 8^3 + 8^3
1027 = 3^3 + 10^3
1027 = 10^3 + 3^3
1064 = 4^3 + 10^3
1064 = 10^3 + 4^3
1072 = 9^3 + 7^3
1072 = 7^3 + 9^3
1125 = 5^3 + 10^3
1125 = 10^3 + 5^3
1216 = 6^3 + 10^3
1216 = 10^3 + 6^3
1241 = 9^3 + 8^3
1241 = 8^3 + 9^3
1331 = 0^3 + 11^3
1331 = 11^3 + 0^3
1332 = 1^3 + 11^3
1332 = 11^3 + 1^3
1339 = 2^3 + 11^3
1339 = 11^3 + 2^3
1343 = 7^3 + 10^3
1343 = 10^3 + 7^3
1358 = 3^3 + 11^3
1358 = 11^3 + 3^3
1395 = 4^3 + 11^3
1395 = 11^3 + 4^3
1456 = 5^3 + 11^3
1456 = 11^3 + 5^3
1458 = 9^3 + 9^3
1512 = 8^3 + 10^3
1512 = 10^3 + 8^3
1547 = 6^3 + 11^3
1547 = 11^3 + 6^3
1674 = 7^3 + 11^3
1674 = 11^3 + 7^3
1728 = 12^3 + 0^3
1728 = 0^3 + 12^3
1729 = 1^3 + 12^3
1729 = 12^3 + 1^3
1729 = 10^3 + 9^3
1729 = 9^3 + 10^3
1736 = 12^3 + 2^3
1736 = 2^3 + 12^3
1755 = 3^3 + 12^3
1755 = 12^3 + 3^3
1792 = 4^3 + 12^3
1792 = 12^3 + 4^3
1843 = 8^3 + 11^3
1843 = 11^3 + 8^3
1853 = 5^3 + 12^3
1853 = 12^3 + 5^3
1944 = 6^3 + 12^3
1944 = 12^3 + 6^3
2000 = 10^3 + 10^3
2060 = 11^3 + 9^3
2060 = 9^3 + 11^3
2071 = 7^3 + 12^3
2071 = 12^3 + 7^3
2240 = 8^3 + 12^3
2240 = 12^3 + 8^3
2331 = 11^3 + 10^3
2331 = 10^3 + 11^3
2457 = 9^3 + 12^3
2457 = 12^3 + 9^3
2662 = 11^3 + 11^3
2728 = 10^3 + 12^3
2728 = 12^3 + 10^3
3059 = 11^3 + 12^3
3059 = 12^3 + 11^3
3456 = 12^3 + 12^3
